src/App.tsx

```diff
@@ -1,61 +1,123 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const maxTodoid = todos
+  .map(todo => todo.id)
+  .reduce((a, b) => Math.max(a, b), 0) + 1;
```

Comment:
this value still calculates just one time. And each new todo will be with id bigger than 1 that its value (so with the same id)

all new todos with key 16:
<img width="498" alt="Screenshot 2022-09-26 at 22 20 08" src="https://user-images.githubusercontent.com/47606379/192361821-7cfecbcf-2349-4289-b3a9-bed58180f09e.png">